src/dbnode/storage/series/series.go

```diff
@@ -290,6 +286,16 @@ func (s *dbSeries) Write(
 	wOpts WriteOptions,
 ) (bool, error) {
 	s.Lock()
+	matchUniqueIndex := wOpts.MatchUniqueIndex
+	if matchUniqueIndex && wOpts.MatchUniqueIndexValue != s.uniqueIndex {
```

Comment:
As discussed previously, fail out here if it's an invalid value?

Reply:
I'm going to make sure that unique index is always > 0 when valid.